File: Model/Pet.cs
using System;
using System.ComponentModel;

public class Pet : INotifyPropertyChanged
{
    private int id;
    private string name;
    private string contacts;
    private string type;
    private string sex;
    private string age;
    private string breed;
    private string weight;
    private Uri picSource;
    private string history;
    private string healthState;
    private string diagnostic;
    private string treatmentPlan;
    private string num_mic;


    public int Id
    {
        get { return id; }
        set
        {
            if (id != value)
            {
                id = value;
                OnPropertyChanged(nameof(Id));
            }
        }
    }

    public string Name
    {
        get { return name; }
        set
        {
            if (name != value)
            {
                name = value;
                OnPropertyChanged(nameof(Name));
            }
        }
    }

    public string Contacts
    {
        get { return contacts; }
        set
        {
            if (contacts != value)
            {
                contacts = value;
                OnPropertyChanged(nameof(Contacts));
            }
        }
    }

    public string Type
    {
        get { return type; }
        set
        {
            if (type != value)
            {
                type = value;
                OnPropertyChanged(nameof(Type));
            }
        }
    }

    public string Sex
    {
        get { return sex; }
        set
        {
            if (sex != value)
            {
                sex = value;
                OnPropertyChanged(nameof(Sex));
            }
        }
    }

    public string Age
    {
        get { return age; }
        set
        {
            if (age != value)
            {
                age = value;
                OnPropertyChanged(nameof(Age));
            }
        }
    }

    public string Breed
    {
        get { return breed; }
        set
        {
            if (breed != value)
            {
                breed = value;
                OnPropertyChanged(nameof(Breed));
            }
        }
    }

    public string Weight
    {
        get { return weight; }
        set
        {
            if (weight != value)
            {
                weight = value;
                OnPropertyChanged(nameof(Weight));
            }
        }
    }

    public Uri PicSource
    {
        get { return picSource; }
        set
        {
            if (picSource != value)
            {
                picSource = value;
                OnPropertyChanged(nameof(PicSource));
            }
        }
    }

    public string History
    {
        get { return history; }
        set {
            if (history != value)
            {
                history = value;
                OnPropertyChanged(nameof(History));
            }
        }
    }

    public string HealthState
    {
        get { return healthState; }
        set
        {
            if (healthState != value)
            {
                healthState = value;
                OnPropertyChanged(nameof(HealthState));
            }
        }
    }
    public string Diagnostic
    {
        get { return diagnostic; }
        set
        {
            if (diagnostic != value)
            {
                diagnostic = value;
                OnPropertyChanged(nameof(Diagnostic));
            }
        }
    }
    public string TreatmentPlan
    {
        get { return treatmentPlan; }
        set
        {
            if (treatmentPlan != value)
            {
                treatmentPlan = value;
                OnPropertyChanged(nameof(TreatmentPlan));
            }
        }
    }

    public string Num_mic
    {
        get { return num_mic; }
        set
        {
            if (num_mic != value)
            {
                num_mic = value;
                OnPropertyChanged(nameof(Num_mic));
            }
        }
    }

    public event PropertyChangedEventHandler PropertyChanged;

    protected virtual void OnPropertyChanged(string propertyName)
    {
      
        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));

    }

}

File: VDC-Electron-React/src/models/Pet.ts
// Corresponds to the original Model/Pet.cs and Swagger definition models.Pet

export interface Pet {
  // Fields from backend models.Pet
  id: number;
  name: string;
  animal_type: string; // From backend (maps to original 'type')
  gender: string;      // From backend (maps to original 'sex')
  age: number;         // Changed to number
  weight?: number;     // Changed to number, optional
  behavior?: string;   // New from backend, optional
  condition?: string;  // New from backend, optional
  research_status?: string; // New from backend, optional

  // Fields kept from original frontend model (might be populated differently now)
  breed: string;       // Kept from original, not in backend Pet model? Check OutputPetDTO or other sources. Assuming needed for display.
  contacts?: string;   // Kept from original, optional
  picSource?: string;  // Kept from original, optional
  history?: string;    // Kept from original, optional (May relate to MedicalEntry)
  healthState?: string;// Kept from original, optional (May relate to MedicalEntry)
  diagnostic?: string; // Kept from original, optional (May relate to MedicalEntry)
  treatmentPlan?: string; // Kept from original, optional (May relate to MedicalEntry)
  num_mic?: string;    // Kept from original, optional
}

// Compatibility aliases removed. Components should use animal_type and gender directly.
// --- Implementation for Compatibility Aliases (if used) ---
// This requires changing the interface to a class or using prototype manipulation.
// Simpler approach for now: Update components directly to use new field names.
// If aliases are strongly desired, we'd need a class implementation:
/*
export class PetClass implements Pet {
  id: number;
  name: string;
  animal_type: string;
  gender: string;
  age: number;
  breed: string;
  weight?: number;
  behavior?: string;
  condition?: string;
  research_status?: string;
  contacts?: string;
  picSource?: string;
  history?: string;
  healthState?: string;
  diagnostic?: string;
  treatmentPlan?: string;
  num_mic?: string;

  constructor(data: Partial<Pet>) {
    // Initialize properties...
    this.id = data.id ?? 0;
    this.name = data.name ?? '';
    this.animal_type = data.animal_type ?? data.type ?? ''; // Handle mapping
    this.gender = data.gender ?? data.sex ?? ''; // Handle mapping
    this.age = typeof data.age === 'string' ? parseInt(data.age, 10) : data.age ?? 0; // Handle potential string age
    this.breed = data.breed ?? '';
    // ... initialize other properties
  }

  get type(): string { return this.animal_type; }
  set type(value: string) { this.animal_type = value; }
  get sex(): string { return this.gender; }
  set sex(value: string) { this.gender = value; }
}
*/
// For now, we'll stick with the interface and update components directly.
